lib/printer.js

```diff
@@ -556,11 +561,18 @@ function genericPrintNoParens(path, options, print) {
                 // both (all) of the holes.
                 parts.push(",");
             } else {
-                if (i > 0)
-                    parts.push(" ");
-                parts.push(print(elemPath));
-                if (i < len - 1)
+                var lines = print(elemPath);
```

Comment:
@benjamn this now prints items in an array twice – is there a way to reuse the printed data from above? If not, I think it is worth it to iterate over the elements twice: once to check if there are holes in the array and once for printing. It would be a little bit more code though.